### PR TITLE
Finish removing relative imports in the front-end

### PR DESCRIPTION
**Pull request by romain-growthbook**
This PR finishes removing relative imports from the front-end. It also turns the corresponding `eslint` rule on for these files.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -106,6 +107,14 @@
         "import/no-named-as-default": "off"
       }
     },
+    {
+      "files": [
+        "./packages/front-end/**/*.ts*"
+      ],
+      "rules": {
+        "no-restricted-imports": ["error", {"patterns": ["..*"]}]
```
Comment (bttf) [inline .eslintrc:115]: I think this might be too restrictive but not sure what the team thinks. Would be great to bring up at the eng. roundtable

**Review by bttf — COMMENTED** [review]
Looks good, one suggestion for discussion across team. Would lean on testing via preview environments generated for the PR to test (see comments for auto-generated link)

File: .eslintrc (hunk 1 above)
Comment (romain-growthbook) [reply .eslintrc:115]: Will do!

File: packages/front-end/tsconfig.json [hunk 2]
```diff
@@ -17,7 +17,9 @@
     "incremental": true,
     "baseUrl": "./",
     "paths": {
-      "@/*": ["./*"]
+      "@/*": ["./*"],
+      "@back-end/*": ["../back-end/*"],
+      "@shared/*": ["../shared/*"]
```
Comment (bttf) [inline packages/front-end/tsconfig.json:22]: I'm not sure if this is correct ... We want to import from `'shared/*'` and not `../shared/*` if I'm not mistaken. The former is package reference (see frontend package.json dependencies  for `shared`), which should reference the compiled TS in your local dev. The latter is relative, and I'm not sure if it works the same or not. (Correct me if I'm wrong please!)

Comment (romain-growthbook) [reply packages/front-end/tsconfig.json]: Work. I think you're right. Have removed the ts path config and substituted all `@shared` for `shared`.

**Review by bttf — APPROVED** [review]

**Pull request merged**